config/dashboard.variables.veneers.common.yaml

```diff
@@ -0,0 +1,162 @@
+      by_name: VariableModel
+      as: ConstantVariable
+      exclude_options: [type]
+  - initialize:
+      by_name: ConstantVariable
+      set:
+        - { property: type, value: constant }
+
+  - duplicate:
+      by_name: VariableModel
+      as: DatasourceVariable
+      exclude_options: [type]
+  - initialize:
+      by_name: DatasourceVariable
+      set:
+        - { property: type, value: datasource }
+
+  - duplicate:
+      by_name: VariableModel
+      as: IntervalVariable
+      exclude_options: [type]
+  - initialize:
+      by_name: IntervalVariable
+      set:
+        - { property: type, value: interval }
+
+  - duplicate:
+      by_name: VariableModel
+      as: TextBoxVariable
+      exclude_options: [type]
+  - initialize:
+      by_name: TextBoxVariable
+      set:
+        - { property: type, value: textbox }
+
+  - duplicate:
+      by_name: VariableModel
+      as: CustomVariable
+      exclude_options: [type]
+  - initialize:
+      by_name: CustomVariable
+      set:
+        - { property: type, value: custom }
+
+  # We don't need this generic builder anymore
+  - omit: { by_name: VariableModel }
+
+options:
+  #################
+  # VariableModel #
+  #################
+  - promote_to_constructor: { by_name: VariableModel.name }
+
+  #################
+  # QueryVariable #
+  #################
+  - omit: { by_builder: QueryVariable.skipUrlSync }
+
+  #################
+  # AdHocVariable #
+  #################
+  - omit: { by_builder: AdHocVariable.skipUrlSync }
+  - omit: { by_builder: AdHocVariable.query }
+  - omit: { by_builder: AdHocVariable.current }
+  - omit: { by_builder: AdHocVariable.multi }
+  - omit: { by_builder: AdHocVariable.options }
+  - omit: { by_builder: AdHocVariable.refresh }
+  - omit: { by_builder: AdHocVariable.sort }
+
+  ####################
+  # ConstantVariable #
+  ####################
+  - omit: { by_builder: ConstantVariable.skipUrlSync }
+  - omit: { by_builder: ConstantVariable.datasource }
+  - omit: { by_builder: ConstantVariable.multi }
+  - omit: { by_builder: ConstantVariable.sort }
+  - omit: { by_builder: ConstantVariable.refresh }
+  - omit: { by_builder: ConstantVariable.options }
+  - omit: { by_builder: ConstantVariable.current }
+  - omit: { by_builder: ConstantVariable.hide }
+  - rename:
+      by_builder: ConstantVariable.query
+      as: value
+
+  ######################
+  # DatasourceVariable #
+  ######################
+  - omit: { by_builder: DatasourceVariable.skipUrlSync }
+  - omit: { by_builder: DatasourceVariable.datasource }
+  - omit: { by_builder: DatasourceVariable.sort }
+  - omit: { by_builder: DatasourceVariable.refresh }
+  - omit: { by_builder: DatasourceVariable.options }
+  - rename:
+      by_builder: DatasourceVariable.query
+      as: datasourceType
+  # DatasourceType("prometheus") instead of explicit DatasourceType(StringOrAny{String: "prometheus"})
+  - struct_fields_as_arguments:
+      by_builder: DatasourceVariable.datasourceType
```

Comment:
Should be better `name` or `type`?. Adding datasource its a bit redundant.

Reply:
Fair point, I'll change that to `type` :)